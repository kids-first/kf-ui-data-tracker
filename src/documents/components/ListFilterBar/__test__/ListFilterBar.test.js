import React from 'react';
import wait from 'waait';
import {MemoryRouter} from 'react-router-dom';
import {MockedProvider} from '@apollo/react-testing';
import {render, cleanup, act, fireEvent} from '@testing-library/react';
import ListFilterBar from '../ListFilterBar';
import {mocks} from '../../../../../__mocks__/kf-api-study-creator/mocks';
import StudyFilesListView from '../../../views/StudyFilesListView';

afterEach(cleanup);

it('renders ListFilterBar with no files', async () => {
  const filters = {
    sortMethod: '',
    sortDirection: 'ascending',
    typeFilterStatus: '',
    tagFilterStatus: '',
    searchString: '',
  };
  const tree = render(
    <MemoryRouter>
      <MockedProvider>
        <ListFilterBar filters={filters} setFilters={jest.fn()} />
      </MockedProvider>
    </MemoryRouter>,
  );
  expect(tree.container).toMatchSnapshot();
});

it('renders ListFilterBar with files -- screen width 1200', async () => {
  const tree = render(
    <MockedProvider mocks={mocks}>
      <MemoryRouter initialEntries={['/study/SD_8WX8QQ06']}>
        <StudyFilesListView match={{params: {kfId: 'SD_8WX8QQ06'}}} />
      </MemoryRouter>
    </MockedProvider>,
  );
  Object.defineProperty(window, 'innerWidth', {
    writable: true,
    configurable: true,
    value: 1200,
  });
  await wait();
  expect(tree.container).toMatchSnapshot();
  act(() => {
    fireEvent.click(tree.getByText(/Date option/i));
  });
  await wait();
  act(() => {
    fireEvent.click(tree.getByText(/Modified date/i));
  });
  await wait();
  act(() => {
    fireEvent.click(tree.getByTestId('sort-direction-button'));
  });
  await wait();
  expect(tree.container).toMatchSnapshot();
  act(() => {
    fireEvent.click(tree.getByTestId('sort-direction-button'));
  });
  await wait();
  expect(tree.container).toMatchSnapshot();
  act(() => {
    fireEvent.click(tree.getByText(/Document type/i));
  });
  await wait();
  act(() => {
    fireEvent.click(tree.getAllByText(/Biospecimen Manifest/i)[1]);
  });
  await wait();
  expect(tree.container).toMatchSnapshot();
  act(() => {
    fireEvent.click(tree.getAllByText(/Tag/i)[0]);
  });
  act(() => {
    fireEvent.click(tree.getAllByText(/dbGaP/i)[0]);
  });
  await wait();
  expect(tree.container).toMatchSnapshot();
  expect(tree.queryAllByTestId('file-item').length).toBe(1);
  act(() => {
    fireEvent.change(tree.getByLabelText('file-search-input'), {
      target: {value: 'its.mp3'},
    });
  });
  expect(tree.container).toMatchSnapshot();
  expect(tree.queryAllByTestId('file-item').length).toBe(1);
});

it('renders ListFilterBar with files -- screen width 800', async () => {
  const tree = render(
    <MockedProvider mocks={mocks}>
      <MemoryRouter initialEntries={['/study/SD_8WX8QQ06']}>
        <StudyFilesListView match={{params: {kfId: 'SD_8WX8QQ06'}}} />
      </MemoryRouter>
    </MockedProvider>,
  );
  Object.defineProperty(window, 'innerWidth', {
    writable: true,
    configurable: true,
    value: 800,
  });
  await wait();
  expect(tree.container).toMatchSnapshot();
  act(() => {
    fireEvent.click(tree.getByText(/Date option/i));
  });
  await wait();
  act(() => {
    fireEvent.click(tree.getByText(/Modified date/i));
  });
  await wait();
  act(() => {
    fireEvent.click(tree.getByTestId('sort-direction-button'));
  });
  await wait();
  expect(tree.container).toMatchSnapshot();
  act(() => {
    fireEvent.click(tree.getByTestId('sort-direction-button'));
  });
  await wait();
  expect(tree.container).toMatchSnapshot();
  act(() => {
    fireEvent.click(tree.getByText(/Document type/i));
  });
  await wait();
  act(() => {
    fireEvent.click(tree.getAllByText(/Biospecimen Manifest/i)[1]);
  });
  await wait();
  expect(tree.container).toMatchSnapshot();
  act(() => {
    fireEvent.click(tree.getAllByText(/Tag/i)[0]);
  });
  act(() => {
    fireEvent.click(tree.getAllByText(/dbGaP/i)[0]);
  });
  await wait();
  expect(tree.container).toMatchSnapshot();
  expect(tree.queryAllByTestId('file-item').length).toBe(1);
  act(() => {
    fireEvent.change(tree.getByLabelText('file-search-input'), {
      target: {value: 'its.mp3'},
    });
  });
  expect(tree.container).toMatchSnapshot();
  expect(tree.queryAllByTestId('file-item').length).toBe(1);
});

it('renders ListFilterBar with files -- screen width 600', async () => {
  const tree = render(
    <MockedProvider mocks={mocks}>
      <MemoryRouter initialEntries={['/study/SD_8WX8QQ06']}>
        <StudyFilesListView match={{params: {kfId: 'SD_8WX8QQ06'}}} />
      </MemoryRouter>
    </MockedProvider>,
  );
  Object.defineProperty(window, 'innerWidth', {
    writable: true,
    configurable: true,
    value: 600,
  });

  await wait();
  expect(tree.container).toMatchSnapshot();
  // Sort files by create date by clicking on the file sorting dropdown
  act(() => {
    fireEvent.click(tree.getByTestId('show-sort-button'));
  });
  act(() => {
    fireEvent.click(tree.getByText(/Date option/i));
  });
  await wait();
  act(() => {
    fireEvent.click(tree.getByText(/Modified date/i));
  });
  await wait();
  act(() => {
    fireEvent.click(tree.getByTestId('sort-direction-button'));
  });
  await wait();
  expect(tree.container).toMatchSnapshot();
  act(() => {
    fireEvent.click(tree.getByTestId('sort-direction-button'));
  });
  await wait();
  expect(tree.container).toMatchSnapshot();
  act(() => {
    fireEvent.click(tree.getByTestId('show-filter-button'));
  });
  expect(tree.container).toMatchSnapshot();
  act(() => {
    fireEvent.click(tree.getByText(/Document type/i));
  });
  await wait();
  act(() => {
    fireEvent.click(tree.getAllByText(/Biospecimen Manifest/i)[1]);
  });
  await wait();
  expect(tree.container).toMatchSnapshot();
  act(() => {
    fireEvent.click(tree.getAllByText(/Tag/i)[0]);
  });
  expect(tree.container).toMatchSnapshot();
  act(() => {
    fireEvent.click(tree.getAllByText(/dbGaP/i)[0]);
  });
  await wait();
  expect(tree.container).toMatchSnapshot();
  expect(tree.queryAllByTestId('file-item').length).toBe(1);
  act(() => {
    fireEvent.change(tree.getByLabelText('file-search-input'), {
      target: {value: 'its.mp3'},
    });
  });
  expect(tree.container).toMatchSnapshot();
  expect(tree.queryAllByTestId('file-item').length).toBe(1);
});
