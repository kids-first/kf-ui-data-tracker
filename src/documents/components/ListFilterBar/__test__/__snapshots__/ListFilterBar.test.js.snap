// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ListFilterBar with files 1`] = `
<div>
  <div
    class="ui basic clearing segment noHorizontalPadding"
  >
    <div
      class="ui basic compact left floated segment noMargin noVerticalPadding noHorizontalPadding"
    >
      <span
        class="smallLabel"
      >
        Filter by:
      </span>
      <div
        aria-expanded="false"
        class="ui selection dropdown"
        role="listbox"
        tabindex="0"
      >
        <div
          aria-live="polite"
          class="default text"
          role="alert"
        >
          Approval status
        </div>
        <i
          aria-hidden="true"
          class="dropdown icon"
        />
        <div
          class="menu transition"
        >
          <div
            aria-checked="false"
            aria-selected="true"
            class="selected item"
            role="option"
            style="pointer-events: all;"
          >
            <div
              class="ui orange tiny basic label text"
            >
              Pending review
            </div>
          </div>
          <div
            aria-checked="false"
            aria-selected="false"
            class="item"
            role="option"
            style="pointer-events: all;"
          >
            <div
              class="ui teal tiny basic label text"
            >
              Approved
            </div>
          </div>
          <div
            aria-checked="false"
            aria-selected="false"
            class="item"
            role="option"
            style="pointer-events: all;"
          >
            <div
              class="ui red tiny basic label text"
            >
              Changes needed
            </div>
          </div>
          <div
            aria-checked="false"
            aria-selected="false"
            class="item"
            role="option"
            style="pointer-events: all;"
          >
            <div
              class="ui blue tiny basic label text"
            >
              Processed
            </div>
          </div>
          <div
            aria-checked="false"
            aria-selected="false"
            class="item"
            role="option"
            style="pointer-events: all;"
          >
            <div
              class="ui blue tiny basic label text"
            >
              Updated
            </div>
          </div>
        </div>
      </div>
      <div
        aria-expanded="false"
        class="ui selection dropdown"
        role="listbox"
        tabindex="0"
      >
        <div
          aria-live="polite"
          class="default text"
          role="alert"
        >
          File type
        </div>
        <i
          aria-hidden="true"
          class="dropdown icon"
        />
        <div
          class="menu transition"
        >
          <div
            aria-checked="false"
            aria-selected="true"
            class="selected item"
            role="option"
            style="pointer-events: all;"
          >
            <small
              class="text"
            >
              <i
                aria-hidden="true"
                class="shipping icon"
              />
               Biospecimen Manifest
            </small>
          </div>
          <div
            aria-checked="false"
            aria-selected="false"
            class="item"
            role="option"
            style="pointer-events: all;"
          >
            <small
              class="text"
            >
              <i
                aria-hidden="true"
                class="hospital icon"
              />
               Clinical/Phenotypic Data
            </small>
          </div>
          <div
            aria-checked="false"
            aria-selected="false"
            class="item"
            role="option"
            style="pointer-events: all;"
          >
            <small
              class="text"
            >
              <i
                aria-hidden="true"
                class="dna icon"
              />
               Sequencing Manifest
            </small>
          </div>
          <div
            aria-checked="false"
            aria-selected="false"
            class="item"
            role="option"
            style="pointer-events: all;"
          >
            <small
              class="text"
            >
              <i
                aria-hidden="true"
                class="question icon"
              />
               Other
            </small>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ui basic compact left floated segment noMargin noVerticalPadding noHorizontalPadding"
    >
      <span
        class="smallLabel"
      >
        Sort by:
      </span>
      <div
        aria-expanded="false"
        class="ui selection dropdown"
        role="listbox"
        tabindex="0"
      >
        <div
          aria-live="polite"
          class="default text"
          role="alert"
        >
          Date option
        </div>
        <i
          aria-hidden="true"
          class="dropdown icon"
        />
        <div
          class="menu transition"
        >
          <div
            aria-checked="false"
            aria-selected="true"
            class="selected item"
            role="option"
            style="pointer-events: all;"
          >
            <span
              class="text"
            >
              Create date
            </span>
          </div>
          <div
            aria-checked="false"
            aria-selected="false"
            class="item"
            role="option"
            style="pointer-events: all;"
          >
            <span
              class="text"
            >
              Modified date
            </span>
          </div>
        </div>
      </div>
      <button
        class="ui basic icon button"
        data-testid="sort-direction-button"
      >
        <i
          aria-hidden="true"
          class="sort content ascending icon"
        />
      </button>
    </div>
    <div
      class="ui icon input"
    >
      <input
        type="text"
        value=""
      />
      <i
        aria-hidden="true"
        class="search icon"
      />
    </div>
  </div>
</div>
`;

exports[`renders ListFilterBar with files 2`] = `
<div>
  <div
    class="ui basic clearing segment noHorizontalPadding"
  >
    <div
      class="ui basic compact left floated segment noMargin noVerticalPadding noHorizontalPadding"
    >
      <span
        class="smallLabel"
      >
        Filter by:
      </span>
      <div
        aria-expanded="true"
        class="ui active visible selection dropdown"
        role="listbox"
        tabindex="0"
      >
        <div
          aria-live="polite"
          class="default text"
          role="alert"
        >
          Pending review
        </div>
        <i
          aria-hidden="true"
          class="dropdown icon"
        />
        <div
          class="visible menu transition"
        >
          <div
            aria-checked="false"
            aria-selected="true"
            class="selected item"
            role="option"
            style="pointer-events: all;"
          >
            <div
              class="ui orange tiny basic label text"
            >
              Pending review
            </div>
          </div>
          <div
            aria-checked="false"
            aria-selected="false"
            class="item"
            role="option"
            style="pointer-events: all;"
          >
            <div
              class="ui teal tiny basic label text"
            >
              Approved
            </div>
          </div>
          <div
            aria-checked="false"
            aria-selected="false"
            class="item"
            role="option"
            style="pointer-events: all;"
          >
            <div
              class="ui red tiny basic label text"
            >
              Changes needed
            </div>
          </div>
          <div
            aria-checked="false"
            aria-selected="false"
            class="item"
            role="option"
            style="pointer-events: all;"
          >
            <div
              class="ui blue tiny basic label text"
            >
              Processed
            </div>
          </div>
          <div
            aria-checked="false"
            aria-selected="false"
            class="item"
            role="option"
            style="pointer-events: all;"
          >
            <div
              class="ui blue tiny basic label text"
            >
              Updated
            </div>
          </div>
        </div>
      </div>
      <div
        aria-expanded="false"
        class="ui selection dropdown"
        role="listbox"
        tabindex="0"
      >
        <div
          aria-live="polite"
          class="default text"
          role="alert"
        >
          File type
        </div>
        <i
          aria-hidden="true"
          class="dropdown icon"
        />
        <div
          class="menu transition"
        >
          <div
            aria-checked="false"
            aria-selected="true"
            class="selected item"
            role="option"
            style="pointer-events: all;"
          >
            <small
              class="text"
            >
              <i
                aria-hidden="true"
                class="shipping icon"
              />
               Biospecimen Manifest
            </small>
          </div>
          <div
            aria-checked="false"
            aria-selected="false"
            class="item"
            role="option"
            style="pointer-events: all;"
          >
            <small
              class="text"
            >
              <i
                aria-hidden="true"
                class="hospital icon"
              />
               Clinical/Phenotypic Data
            </small>
          </div>
          <div
            aria-checked="false"
            aria-selected="false"
            class="item"
            role="option"
            style="pointer-events: all;"
          >
            <small
              class="text"
            >
              <i
                aria-hidden="true"
                class="dna icon"
              />
               Sequencing Manifest
            </small>
          </div>
          <div
            aria-checked="false"
            aria-selected="false"
            class="item"
            role="option"
            style="pointer-events: all;"
          >
            <small
              class="text"
            >
              <i
                aria-hidden="true"
                class="question icon"
              />
               Other
            </small>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ui basic compact left floated segment noMargin noVerticalPadding noHorizontalPadding"
    >
      <span
        class="smallLabel"
      >
        Sort by:
      </span>
      <div
        aria-expanded="false"
        class="ui selection dropdown"
        role="listbox"
        tabindex="0"
      >
        <div
          aria-live="polite"
          class="default text"
          role="alert"
        >
          Date option
        </div>
        <i
          aria-hidden="true"
          class="dropdown icon"
        />
        <div
          class="menu transition"
        >
          <div
            aria-checked="false"
            aria-selected="true"
            class="selected item"
            role="option"
            style="pointer-events: all;"
          >
            <span
              class="text"
            >
              Create date
            </span>
          </div>
          <div
            aria-checked="false"
            aria-selected="false"
            class="item"
            role="option"
            style="pointer-events: all;"
          >
            <span
              class="text"
            >
              Modified date
            </span>
          </div>
        </div>
      </div>
      <button
        class="ui basic icon button"
        data-testid="sort-direction-button"
      >
        <i
          aria-hidden="true"
          class="sort content ascending icon"
        />
      </button>
    </div>
    <div
      class="ui icon input"
    >
      <input
        type="text"
        value=""
      />
      <i
        aria-hidden="true"
        class="search icon"
      />
    </div>
  </div>
</div>
`;

exports[`renders ListFilterBar with files 3`] = `
<div>
  <div
    class="ui basic clearing segment noHorizontalPadding"
  >
    <div
      class="ui basic compact left floated segment noMargin noVerticalPadding noHorizontalPadding"
    >
      <span
        class="smallLabel"
      >
        Filter by:
      </span>
      <div
        aria-expanded="false"
        class="ui selection dropdown"
        role="listbox"
        tabindex="0"
      >
        <div
          aria-live="polite"
          class="default text"
          role="alert"
        >
          Approval status
        </div>
        <i
          aria-hidden="true"
          class="dropdown icon"
        />
        <div
          class="menu transition"
        >
          <div
            aria-checked="false"
            aria-selected="true"
            class="selected item"
            role="option"
            style="pointer-events: all;"
          >
            <div
              class="ui orange tiny basic label text"
            >
              Pending review
            </div>
          </div>
          <div
            aria-checked="false"
            aria-selected="false"
            class="item"
            role="option"
            style="pointer-events: all;"
          >
            <div
              class="ui teal tiny basic label text"
            >
              Approved
            </div>
          </div>
          <div
            aria-checked="false"
            aria-selected="false"
            class="item"
            role="option"
            style="pointer-events: all;"
          >
            <div
              class="ui red tiny basic label text"
            >
              Changes needed
            </div>
          </div>
          <div
            aria-checked="false"
            aria-selected="false"
            class="item"
            role="option"
            style="pointer-events: all;"
          >
            <div
              class="ui blue tiny basic label text"
            >
              Processed
            </div>
          </div>
          <div
            aria-checked="false"
            aria-selected="false"
            class="item"
            role="option"
            style="pointer-events: all;"
          >
            <div
              class="ui blue tiny basic label text"
            >
              Updated
            </div>
          </div>
        </div>
      </div>
      <div
        aria-expanded="true"
        class="ui active visible selection dropdown"
        role="listbox"
        tabindex="0"
      >
        <div
          aria-live="polite"
          class="default text"
          role="alert"
        >
          Biospecimen Manifest
        </div>
        <i
          aria-hidden="true"
          class="dropdown icon"
        />
        <div
          class="visible menu transition"
        >
          <div
            aria-checked="false"
            aria-selected="true"
            class="selected item"
            role="option"
            style="pointer-events: all;"
          >
            <small
              class="text"
            >
              <i
                aria-hidden="true"
                class="shipping icon"
              />
               Biospecimen Manifest
            </small>
          </div>
          <div
            aria-checked="false"
            aria-selected="false"
            class="item"
            role="option"
            style="pointer-events: all;"
          >
            <small
              class="text"
            >
              <i
                aria-hidden="true"
                class="hospital icon"
              />
               Clinical/Phenotypic Data
            </small>
          </div>
          <div
            aria-checked="false"
            aria-selected="false"
            class="item"
            role="option"
            style="pointer-events: all;"
          >
            <small
              class="text"
            >
              <i
                aria-hidden="true"
                class="dna icon"
              />
               Sequencing Manifest
            </small>
          </div>
          <div
            aria-checked="false"
            aria-selected="false"
            class="item"
            role="option"
            style="pointer-events: all;"
          >
            <small
              class="text"
            >
              <i
                aria-hidden="true"
                class="question icon"
              />
               Other
            </small>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ui basic compact left floated segment noMargin noVerticalPadding noHorizontalPadding"
    >
      <span
        class="smallLabel"
      >
        Sort by:
      </span>
      <div
        aria-expanded="false"
        class="ui selection dropdown"
        role="listbox"
        tabindex="0"
      >
        <div
          aria-live="polite"
          class="default text"
          role="alert"
        >
          Date option
        </div>
        <i
          aria-hidden="true"
          class="dropdown icon"
        />
        <div
          class="menu transition"
        >
          <div
            aria-checked="false"
            aria-selected="true"
            class="selected item"
            role="option"
            style="pointer-events: all;"
          >
            <span
              class="text"
            >
              Create date
            </span>
          </div>
          <div
            aria-checked="false"
            aria-selected="false"
            class="item"
            role="option"
            style="pointer-events: all;"
          >
            <span
              class="text"
            >
              Modified date
            </span>
          </div>
        </div>
      </div>
      <button
        class="ui basic icon button"
        data-testid="sort-direction-button"
      >
        <i
          aria-hidden="true"
          class="sort content ascending icon"
        />
      </button>
    </div>
    <div
      class="ui icon input"
    >
      <input
        type="text"
        value=""
      />
      <i
        aria-hidden="true"
        class="search icon"
      />
    </div>
  </div>
</div>
`;

exports[`renders ListFilterBar with files 4`] = `
<div>
  <div
    class="ui basic clearing segment noHorizontalPadding"
  >
    <div
      class="ui basic compact left floated segment noMargin noVerticalPadding noHorizontalPadding"
    >
      <span
        class="smallLabel"
      >
        Filter by:
      </span>
      <div
        aria-expanded="false"
        class="ui selection dropdown"
        role="listbox"
        tabindex="0"
      >
        <div
          aria-live="polite"
          class="default text"
          role="alert"
        >
          Approval status
        </div>
        <i
          aria-hidden="true"
          class="dropdown icon"
        />
        <div
          class="menu transition"
        >
          <div
            aria-checked="false"
            aria-selected="true"
            class="selected item"
            role="option"
            style="pointer-events: all;"
          >
            <div
              class="ui orange tiny basic label text"
            >
              Pending review
            </div>
          </div>
          <div
            aria-checked="false"
            aria-selected="false"
            class="item"
            role="option"
            style="pointer-events: all;"
          >
            <div
              class="ui teal tiny basic label text"
            >
              Approved
            </div>
          </div>
          <div
            aria-checked="false"
            aria-selected="false"
            class="item"
            role="option"
            style="pointer-events: all;"
          >
            <div
              class="ui red tiny basic label text"
            >
              Changes needed
            </div>
          </div>
          <div
            aria-checked="false"
            aria-selected="false"
            class="item"
            role="option"
            style="pointer-events: all;"
          >
            <div
              class="ui blue tiny basic label text"
            >
              Processed
            </div>
          </div>
          <div
            aria-checked="false"
            aria-selected="false"
            class="item"
            role="option"
            style="pointer-events: all;"
          >
            <div
              class="ui blue tiny basic label text"
            >
              Updated
            </div>
          </div>
        </div>
      </div>
      <div
        aria-expanded="false"
        class="ui selection dropdown"
        role="listbox"
        tabindex="0"
      >
        <div
          aria-live="polite"
          class="default text"
          role="alert"
        >
          File type
        </div>
        <i
          aria-hidden="true"
          class="dropdown icon"
        />
        <div
          class="menu transition"
        >
          <div
            aria-checked="false"
            aria-selected="true"
            class="selected item"
            role="option"
            style="pointer-events: all;"
          >
            <small
              class="text"
            >
              <i
                aria-hidden="true"
                class="shipping icon"
              />
               Biospecimen Manifest
            </small>
          </div>
          <div
            aria-checked="false"
            aria-selected="false"
            class="item"
            role="option"
            style="pointer-events: all;"
          >
            <small
              class="text"
            >
              <i
                aria-hidden="true"
                class="hospital icon"
              />
               Clinical/Phenotypic Data
            </small>
          </div>
          <div
            aria-checked="false"
            aria-selected="false"
            class="item"
            role="option"
            style="pointer-events: all;"
          >
            <small
              class="text"
            >
              <i
                aria-hidden="true"
                class="dna icon"
              />
               Sequencing Manifest
            </small>
          </div>
          <div
            aria-checked="false"
            aria-selected="false"
            class="item"
            role="option"
            style="pointer-events: all;"
          >
            <small
              class="text"
            >
              <i
                aria-hidden="true"
                class="question icon"
              />
               Other
            </small>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ui basic compact left floated segment noMargin noVerticalPadding noHorizontalPadding"
    >
      <span
        class="smallLabel"
      >
        Sort by:
      </span>
      <div
        aria-expanded="true"
        class="ui active visible selection dropdown"
        role="listbox"
        tabindex="0"
      >
        <div
          aria-live="polite"
          class="default text"
          role="alert"
        >
          Create date
        </div>
        <i
          aria-hidden="true"
          class="dropdown icon"
        />
        <div
          class="visible menu transition"
        >
          <div
            aria-checked="false"
            aria-selected="true"
            class="selected item"
            role="option"
            style="pointer-events: all;"
          >
            <span
              class="text"
            >
              Create date
            </span>
          </div>
          <div
            aria-checked="false"
            aria-selected="false"
            class="item"
            role="option"
            style="pointer-events: all;"
          >
            <span
              class="text"
            >
              Modified date
            </span>
          </div>
        </div>
      </div>
      <button
        class="ui basic icon button"
        data-testid="sort-direction-button"
      >
        <i
          aria-hidden="true"
          class="sort content ascending icon"
        />
      </button>
    </div>
    <div
      class="ui icon input"
    >
      <input
        type="text"
        value=""
      />
      <i
        aria-hidden="true"
        class="search icon"
      />
    </div>
  </div>
</div>
`;

exports[`renders ListFilterBar with files 5`] = `
<div>
  <div
    class="ui basic clearing segment noHorizontalPadding"
  >
    <div
      class="ui basic compact left floated segment noMargin noVerticalPadding noHorizontalPadding"
    >
      <span
        class="smallLabel"
      >
        Filter by:
      </span>
      <div
        aria-expanded="false"
        class="ui selection dropdown"
        role="listbox"
        tabindex="0"
      >
        <div
          aria-live="polite"
          class="default text"
          role="alert"
        >
          Approval status
        </div>
        <i
          aria-hidden="true"
          class="dropdown icon"
        />
        <div
          class="menu transition"
        >
          <div
            aria-checked="false"
            aria-selected="true"
            class="selected item"
            role="option"
            style="pointer-events: all;"
          >
            <div
              class="ui orange tiny basic label text"
            >
              Pending review
            </div>
          </div>
          <div
            aria-checked="false"
            aria-selected="false"
            class="item"
            role="option"
            style="pointer-events: all;"
          >
            <div
              class="ui teal tiny basic label text"
            >
              Approved
            </div>
          </div>
          <div
            aria-checked="false"
            aria-selected="false"
            class="item"
            role="option"
            style="pointer-events: all;"
          >
            <div
              class="ui red tiny basic label text"
            >
              Changes needed
            </div>
          </div>
          <div
            aria-checked="false"
            aria-selected="false"
            class="item"
            role="option"
            style="pointer-events: all;"
          >
            <div
              class="ui blue tiny basic label text"
            >
              Processed
            </div>
          </div>
          <div
            aria-checked="false"
            aria-selected="false"
            class="item"
            role="option"
            style="pointer-events: all;"
          >
            <div
              class="ui blue tiny basic label text"
            >
              Updated
            </div>
          </div>
        </div>
      </div>
      <div
        aria-expanded="false"
        class="ui selection dropdown"
        role="listbox"
        tabindex="0"
      >
        <div
          aria-live="polite"
          class="default text"
          role="alert"
        >
          File type
        </div>
        <i
          aria-hidden="true"
          class="dropdown icon"
        />
        <div
          class="menu transition"
        >
          <div
            aria-checked="false"
            aria-selected="true"
            class="selected item"
            role="option"
            style="pointer-events: all;"
          >
            <small
              class="text"
            >
              <i
                aria-hidden="true"
                class="shipping icon"
              />
               Biospecimen Manifest
            </small>
          </div>
          <div
            aria-checked="false"
            aria-selected="false"
            class="item"
            role="option"
            style="pointer-events: all;"
          >
            <small
              class="text"
            >
              <i
                aria-hidden="true"
                class="hospital icon"
              />
               Clinical/Phenotypic Data
            </small>
          </div>
          <div
            aria-checked="false"
            aria-selected="false"
            class="item"
            role="option"
            style="pointer-events: all;"
          >
            <small
              class="text"
            >
              <i
                aria-hidden="true"
                class="dna icon"
              />
               Sequencing Manifest
            </small>
          </div>
          <div
            aria-checked="false"
            aria-selected="false"
            class="item"
            role="option"
            style="pointer-events: all;"
          >
            <small
              class="text"
            >
              <i
                aria-hidden="true"
                class="question icon"
              />
               Other
            </small>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ui basic compact left floated segment noMargin noVerticalPadding noHorizontalPadding"
    >
      <span
        class="smallLabel"
      >
        Sort by:
      </span>
      <div
        aria-expanded="false"
        class="ui selection dropdown"
        role="listbox"
        tabindex="0"
      >
        <div
          aria-live="polite"
          class="default text"
          role="alert"
        >
          Date option
        </div>
        <i
          aria-hidden="true"
          class="dropdown icon"
        />
        <div
          class="menu transition"
        >
          <div
            aria-checked="false"
            aria-selected="true"
            class="selected item"
            role="option"
            style="pointer-events: all;"
          >
            <span
              class="text"
            >
              Create date
            </span>
          </div>
          <div
            aria-checked="false"
            aria-selected="false"
            class="item"
            role="option"
            style="pointer-events: all;"
          >
            <span
              class="text"
            >
              Modified date
            </span>
          </div>
        </div>
      </div>
      <button
        class="ui basic icon button"
        data-testid="sort-direction-button"
      >
        <i
          aria-hidden="true"
          class="sort content descending icon"
        />
      </button>
    </div>
    <div
      class="ui icon input"
    >
      <input
        type="text"
        value=""
      />
      <i
        aria-hidden="true"
        class="search icon"
      />
    </div>
  </div>
</div>
`;

exports[`renders ListFilterBar with no files 1`] = `
<div>
  <div
    class="ui basic clearing segment noHorizontalPadding"
  >
    <div
      class="ui basic compact left floated segment noMargin noVerticalPadding noHorizontalPadding"
    >
      <span
        class="smallLabel"
      >
        Filter by:
      </span>
      <div
        aria-expanded="false"
        class="ui selection dropdown"
        role="listbox"
        tabindex="0"
      >
        <div
          aria-live="polite"
          class="default text"
          role="alert"
        >
          Approval status
        </div>
        <i
          aria-hidden="true"
          class="dropdown icon"
        />
        <div
          class="menu transition"
        >
          <div
            aria-checked="false"
            aria-selected="true"
            class="selected item"
            role="option"
            style="pointer-events: all;"
          >
            <div
              class="ui orange tiny basic label text"
            >
              Pending review
            </div>
          </div>
          <div
            aria-checked="false"
            aria-selected="false"
            class="item"
            role="option"
            style="pointer-events: all;"
          >
            <div
              class="ui teal tiny basic label text"
            >
              Approved
            </div>
          </div>
          <div
            aria-checked="false"
            aria-selected="false"
            class="item"
            role="option"
            style="pointer-events: all;"
          >
            <div
              class="ui red tiny basic label text"
            >
              Changes needed
            </div>
          </div>
          <div
            aria-checked="false"
            aria-selected="false"
            class="item"
            role="option"
            style="pointer-events: all;"
          >
            <div
              class="ui blue tiny basic label text"
            >
              Processed
            </div>
          </div>
          <div
            aria-checked="false"
            aria-selected="false"
            class="item"
            role="option"
            style="pointer-events: all;"
          >
            <div
              class="ui blue tiny basic label text"
            >
              Updated
            </div>
          </div>
        </div>
      </div>
      <div
        aria-expanded="false"
        class="ui selection dropdown"
        role="listbox"
        tabindex="0"
      >
        <div
          aria-live="polite"
          class="default text"
          role="alert"
        >
          File type
        </div>
        <i
          aria-hidden="true"
          class="dropdown icon"
        />
        <div
          class="menu transition"
        >
          <div
            aria-checked="false"
            aria-selected="true"
            class="selected item"
            role="option"
            style="pointer-events: all;"
          >
            <small
              class="text"
            >
              <i
                aria-hidden="true"
                class="shipping icon"
              />
               Biospecimen Manifest
            </small>
          </div>
          <div
            aria-checked="false"
            aria-selected="false"
            class="item"
            role="option"
            style="pointer-events: all;"
          >
            <small
              class="text"
            >
              <i
                aria-hidden="true"
                class="hospital icon"
              />
               Clinical/Phenotypic Data
            </small>
          </div>
          <div
            aria-checked="false"
            aria-selected="false"
            class="item"
            role="option"
            style="pointer-events: all;"
          >
            <small
              class="text"
            >
              <i
                aria-hidden="true"
                class="dna icon"
              />
               Sequencing Manifest
            </small>
          </div>
          <div
            aria-checked="false"
            aria-selected="false"
            class="item"
            role="option"
            style="pointer-events: all;"
          >
            <small
              class="text"
            >
              <i
                aria-hidden="true"
                class="question icon"
              />
               Other
            </small>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ui basic compact left floated segment noMargin noVerticalPadding noHorizontalPadding"
    >
      <span
        class="smallLabel"
      >
        Sort by:
      </span>
      <div
        aria-expanded="false"
        class="ui selection dropdown"
        role="listbox"
        tabindex="0"
      >
        <div
          aria-live="polite"
          class="default text"
          role="alert"
        >
          Date option
        </div>
        <i
          aria-hidden="true"
          class="dropdown icon"
        />
        <div
          class="menu transition"
        >
          <div
            aria-checked="false"
            aria-selected="true"
            class="selected item"
            role="option"
            style="pointer-events: all;"
          >
            <span
              class="text"
            >
              Create date
            </span>
          </div>
          <div
            aria-checked="false"
            aria-selected="false"
            class="item"
            role="option"
            style="pointer-events: all;"
          >
            <span
              class="text"
            >
              Modified date
            </span>
          </div>
        </div>
      </div>
      <button
        class="ui basic icon button"
        data-testid="sort-direction-button"
      >
        <i
          aria-hidden="true"
          class="sort content ascending icon"
        />
      </button>
    </div>
    <div
      class="ui icon input"
    >
      <input
        type="text"
        value=""
      />
      <i
        aria-hidden="true"
        class="search icon"
      />
    </div>
  </div>
</div>
`;
